src/lj_snap.c

```diff
@@ -599,9 +599,9 @@ static void snap_restoreval(jit_State *J, GCtrace *T, ExitState *ex,
   }
   if (LJ_UNLIKELY(bloomtest(rfilt, ref)))
     rs = snap_renameref(T, snapno, ref, rs);
-  lua_assert(!LJ_GC64);  /* TODO_GC64: handle 64 bit references. */
+  // lua_assert(!LJ_GC64);  /* TODO_GC64: handle 64 bit references. */
   if (ra_hasspill(regsp_spill(rs))) {  /* Restore from spill slot. */
-    int32_t *sps = &ex->spill[regsp_spill(rs)];
+    SnapEntry *sps = &ex->spill[regsp_spill(rs)];
```

Comment:
No. In the non-integer cases, `sps` is already cast to a different pointer type before dereferencing, and in the integer case, we want to dereference an `int32_t` rather than a potentially 64-bit type.